lib/chain.js

```diff
@@ -80,14 +85,18 @@ module.exports = class SetupChain {
     if (path_type === 'string') return this[lookup](this.parse(path))
 
     const out = Object.create(null)
+    this.context.push(out)
     // path is an object remap values with lookup result
     for (const [key, value] of Object.entries(path)) {
 
       // remap array values with lookup result
       if (Array.isArray(value)) {
         out[key] = value.map((val) => {
           if (typeOf(val) === 'object') return this.lookup(val)
-          return this[lookup](this.parse(val))
+          this.context.push(val)
+          const resolved = this[lookup](this.parse(val))
+          this.context.pop()
```

Comment:
is this even necessary for non-object types? I can't see a path where the context pushed/popped here is actually used.

Reply:
Maybe not, but it keeps the order of the stack. other wise I think I would have to keep track of the case where item were not added. Like this its just dumb recursion

Reply:
Oh i see what you mean. don't handle the array specifically, and let the object bits do everything?

Reply:
pretty much what you said - I don't see a code path where an array would actually use the context placed on the stack here